### ISSUE TITLE
typos in NumPy tutorials

### ISSUE THREAD

**Issue by godaygo** [numpy/numpy#8537]
I found some typos while reading tutorials [docs.scipy.org](https://docs.scipy.org). I don`t know of it is a right place to post them...

------------------------------

1.  [Quick Start: Stacking together different arrays](https://docs.scipy.org/doc/numpy/user/quickstart.html#stacking-together-different-arrays)

**WRONG**: The function `column_stack` stacks 1D arrays as columns into a 2D array. It is equivalent to `vstack` only for 1D arrays.

**RIGHT**:   The function `column_stack` stacks 1D arrays as columns into a 2D array. It is equivalent to `transpose(vstack((arr1, arr2)))`  in 1D case and `hstack` in 2D case. 

It will be nice to add that: NumPy has also the `row_stack` function which is simply an alias to the `vstack` function. 

----

2. [NumPy Basics: Indexing Multi-dimensional arrays¶](https://docs.scipy.org/doc/numpy/user/basics.indexing.html#indexing-multi-dimensional-arrays)

```python
>>> y[np.array([0,2,4])]
array([[ 0,  1,  2,  3,  4,  5,  6],
       [14, 15, 16, 17, 18, 19, 20],
       [28, 29, 30, 31, 32, 33, 34]])
```
**WRONG**: What results is the construction of a new array where each value of the index array selects one row from the array being indexed and the resultant array has the resulting shape (size of row, number index elements).

**RIGHT**: What results is the construction of a new array where each value of the index array selects one row from the array being indexed and the resultant array has the resulting shape **(number index elements, size of row)**.

----

3. [NumPy Reference: Scalars: Indexing](https://docs.scipy.org/doc/numpy/reference/arrays.scalars.html#indexing)

**WRONG**: `x[()]` returns a 0-dimensional `ndarray`.

**RIGHT** `x[...]` returns a 0-dimensional `ndarray`.

One more comment why:
```python
>>> a = np.array(1)
>>> b = np.int_(1)

>>> a[...] 
array(1)
>>> b[...] 
array(1)

>>> a[()] 
1
>>> b[()] 
1
```
Should this operations be *reversed* for `a` and `b`,  is this a right behavior?

----

4. [NumPy Reference:Indexing](https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html)

**??**: The basic slice syntax is `i:j:k` where `i` is the starting index, `j` is the stopping index, and `k` is the step (`k != 0`). This selects the `m` elements (in the corresponding dimension) with index values `i`, `i + k`, ..., `i + (m - 1) k` where `m = q + (r !=0 )` and `q` and `r` are the quotient and remainder obtained by dividing `j - i` by `k`:` j - i = q k + r`, so that `i + (m - 1) k < j`.

**I'm curious, if it is really so hard math (`m = q + (r !=0 ))`?**

--------------

**Comment by seberg** [comment]
Its fine to report here. If you have some time, it would be awesome if you can create a pull request (if you take care even the edit button in github works). All those files are somewhere in `numpy/doc/source/...`. If you create a pull request, try to commit it with something like `DOC: short explenation...`

**Comment by godaygo** [comment]
@seberg  I'm a complete noob at all this (and never did PR before), but I will try:)

**Comment by charris** [comment]
Closed by #8575 .
